update/process/processPending.go

```diff
@@ -0,0 +1,186 @@
+func NewPendingTransactionProcessor(args ArgsPendingTransactionProcessor) (*pendingProcessor, error) {
+	if check.IfNil(args.Accounts) {
+		return nil, update.ErrNilAccounts
+	}
+	if check.IfNil(args.TxProcessor) {
+		return nil, update.ErrNilTxProcessor
+	}
+	if check.IfNil(args.ScrTxProcessor) {
+		return nil, update.ErrNilSCRProcessor
+	}
+	if check.IfNil(args.RwdTxProcessor) {
+		return nil, update.ErrNilRwdTxProcessor
+	}
+	if check.IfNil(args.PubKeyConv) {
+		return nil, update.ErrNilPubkeyConverter
+	}
+	if check.IfNil(args.ShardCoordinator) {
+		return nil, update.ErrNilShardCoordinator
+	}
+
+	return &pendingProcessor{
+		accounts:         args.Accounts,
+		txProcessor:      args.TxProcessor,
+		rwdTxProcessor:   args.RwdTxProcessor,
+		scrTxProcessor:   args.ScrTxProcessor,
+		pubKeyConv:       args.PubKeyConv,
+		shardCoordinator: args.ShardCoordinator,
+	}, nil
+}
+
+// ProcessTransactionsDstMe processes all the transactions in which destination is the current shard
+func (p *pendingProcessor) ProcessTransactionsDstMe(mapTxs map[string]data.TransactionHandler) (block.MiniBlockSlice, error) {
+	sortedTxs := make([]*txInfo, 0, len(mapTxs))
+	for hash, tx := range mapTxs {
+		sortedTxs = append(sortedTxs, &txInfo{
+			hash: hash,
+			tx:   tx,
+		})
+	}
+
+	sort.Slice(sortedTxs, func(i, j int) bool {
+		return sortedTxs[i].hash < sortedTxs[j].hash
+	})
+
+	mapMiniBlocks := make(map[string]*block.MiniBlock)
+	for _, info := range sortedTxs {
+		rcvAddress, err := p.pubKeyConv.CreateAddressFromBytes(info.tx.GetRcvAddr())
+		if err != nil {
+			continue
+		}
+
+		sndAddress, err := p.pubKeyConv.CreateAddressFromBytes(info.tx.GetSndAddr())
+		if err != nil {
+			continue
+		}
+
+		sndShardId := p.shardCoordinator.ComputeId(sndAddress)
+
+		dstShardID := p.shardCoordinator.ComputeId(rcvAddress)
+		if dstShardID != p.shardCoordinator.SelfId() {
+			continue
+		}
+
+		blockType, err := p.processSingleTransaction(info)
+		if err != nil {
+			continue
+		}
+
+		localID := fmt.Sprintf("%d%d%d", sndShardId, dstShardID, blockType)
+		mb, ok := mapMiniBlocks[localID]
+		if !ok {
+			mb = &block.MiniBlock{
+				TxHashes:        make([][]byte, 0),
+				ReceiverShardID: dstShardID,
+				SenderShardID:   sndShardId,
+				Type:            blockType,
+			}
+		}
+
+		mb.TxHashes = append(mb.TxHashes, []byte(info.hash))
+	}
+
+	_, err := p.accounts.Commit()
+	if err != nil {
+		return nil, err
+	}
+
+	miniBlocks := make(block.MiniBlockSlice, 0, len(mapMiniBlocks))
+	for _, mb := range mapMiniBlocks {
+		miniBlocks = append(miniBlocks, mb)
+	}
+
+	sort.Slice(miniBlocks, func(i, j int) bool {
+		if miniBlocks[i].SenderShardID == miniBlocks[j].SenderShardID {
+			return miniBlocks[i].Type < miniBlocks[j].Type
+		}
+		return miniBlocks[i].SenderShardID < miniBlocks[j].SenderShardID
+	})
+
+	return miniBlocks, nil
+}
+
+func (p *pendingProcessor) processSingleTransaction(info *txInfo) (block.Type, error) {
+	rwdTx, ok := info.tx.(*rewardTx.RewardTx)
```

Comment:
this is not quite extensible without modifying the code. I like the pattern found in the transaction coordinator that takes the pre/intermediate processor based on the tx block type and then send the transaction to that preprocessor.

Reply:
these processors do not implement the same interface - preprocessors and postprocessors do much more then needed here. I will think about having a container plus same inteerface, but needs a lot of change in production code. Maybe as a separate task to make all these processors to implement the same ProcessTransaction interface ?